common/pinot/response_utility.go

```diff
@@ -64,33 +67,44 @@ type VisibilityRecord struct {
 	ShardID       int16
 }
 
-func ConvertSearchResultToVisibilityRecord(hit []interface{}, columnNames []string, logger log.Logger) *p.InternalVisibilityWorkflowExecutionInfo {
+func ConvertSearchResultToVisibilityRecord(hit []interface{}, columnNames []string) (*p.InternalVisibilityWorkflowExecutionInfo, error) {
 	if len(hit) != len(columnNames) {
-		return nil
+		return nil, fmt.Errorf("length of hit (%v) is not equal with length of columnNames(%v)", len(hit), len(columnNames))
```

Comment:
What if we have upgraded the pinot schema but haven't deployed code changes, will it fail here?

Reply:
Probably not in this diff, but should have a version to guard here, also backward compatible cc: @davidporter-id-au

Reply:
Ideally those will always be equal since both things are from the same piece pinot response. If those are not equal, then it should be an error from the pinot side not us.

Reply:
Oh, this schema also comes from the Pinot response, not reading from config, correct?

Reply:
correct